From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hardened NanoKVM kernel port <noreply@local>
Date: Sun, 23 Aug 2026 22:44:43 +0300
Subject: [PATCH 8/9] ion: use stable close_fd helper

Adapt the CVITEK exported ion_free compatibility hook to the post-5.10.4 internal close API.
---
 drivers/staging/android/ion/ion.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/android/ion/ion.c b/drivers/staging/android/ion/ion.c
index e78e203001..9d0d95e263 100644
--- a/drivers/staging/android/ion/ion.c
+++ b/drivers/staging/android/ion/ion.c
@@ -10,6 +10,7 @@
 #include <linux/dma-buf.h>
 #include <linux/err.h>
 #include <linux/export.h>
+#include <linux/fdtable.h>
 #include <linux/file.h>
 #include <linux/freezer.h>
 #include <linux/fs.h>
@@ -906,10 +907,9 @@ static int ion_device_create(void)
 }
 subsys_initcall(ion_device_create);
 #ifdef CONFIG_ION_CVITEK
-#include <linux/syscalls.h>
 void ion_free(pid_t pid, int fd)
 {
-	ksys_close(fd);
+	close_fd(fd);
 }
 EXPORT_SYMBOL(ion_free);
 #endif
